src/main/java/vg/civcraft/mc/civmodcore/api/NamespaceAPI.java

```diff
@@ -0,0 +1,59 @@
+package vg.civcraft.mc.civmodcore.api;
+
+import com.google.common.base.Strings;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+
+/**
+ * Utility class to make dealing with namespace keys easier.
+ */
+public final class NamespaceAPI {
+
+	/**
+	 * Converts a stringified namespace back into a {@link NamespacedKey}.
+	 *
+	 * @param key The stringified namespace, which MUST be formatted as {@code namespace:name}
+	 * @return Returns a valid {@link NamespacedKey}, or null.
+	 *
+	 * @exception IllegalArgumentException Will throw if the stringified key fails a "[a-z0-9._-]+" check, or if the
+	 *     total length is longer than 256.
+	 */
+	@SuppressWarnings("deprecation")
+	public static NamespacedKey fromString(String key) {
```

Comment:
Don't add a method and instantly deprecate it, just don't have it and tell people to git gud

Reply:
I didn't D:

The `NamespacedKey`'s constructor that takes a namespace and a key is deprecated despite it being used internally for `NamespacedKey.minecraft(String)` and `NamespacedKey.randomKey()`. I'm suppressing the deprecation warning, not deprecating the API.

Reply:
Oh wait i thought it was an `@Deprecated` annotation. Nevermind, carry on.